src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs

```diff
@@ -57,18 +60,29 @@ public static FormulaValue FromJson(JsonElement element)
                     return new BooleanValue(IRContext.NotInSource(FormulaType.Boolean), true);
 
                 case JsonValueKind.Object:
-                    return RecordFromJsonObject(element);
+                    return RecordFromJsonObject(element, formulaType as RecordType);
 
                 case JsonValueKind.Array:
-                    return TableFromJsonArray(element);
+                    if (formulaType == null || formulaType is TableType)
+                    {
+                        return TableFromJsonArray(element, formulaType as TableType);
+                    }
+                    else if (formulaType is UntypedObjectType uot)
+                    {
+                        return new UntypedObjectValue(IRContext.NotInSource(FormulaType.UntypedObject), new JsonUntypedObject(element));
+                    }
```

Comment:
Should this case be top-level?
IE, regardless of the element kind, if they ask for an UntypedObject, we could provide it.

Reply:
It should be at top level, you're right